int32 commandCode
float32 distance
float32 angle
int32 location
int32 numTimes
string commandText
